main/php_ini_builder.c

```diff
@@ -0,0 +1,82 @@
+/*
+   +----------------------------------------------------------------------+
+   | Copyright (c) The PHP Group                                          |
+   +----------------------------------------------------------------------+
+   | This source file is subject to version 3.01 of the PHP license,      |
+   | that is bundled with this package in the file LICENSE, and is        |
+   | available through the world-wide-web at the following url:           |
+   | https://www.php.net/license/3_01.txt                                 |
+   | If you did not receive a copy of the PHP license and are unable to   |
+   | obtain it through the world-wide-web, please send a note to          |
+   | license@php.net so we can mail you a copy immediately.               |
+   +----------------------------------------------------------------------+
+   | Authors: Max Kellermann <max.kellermann@ionos.com>                   |
+   +----------------------------------------------------------------------+
+*/
+
+#include "php_ini_builder.h"
+
+#include <ctype.h>
+#include <string.h>
+
+PHPAPI void php_ini_builder_prepend(struct php_ini_builder *b, const char *src, size_t length)
+{
+	php_ini_builder_realloc(b, length);
+	if (b->length > 0)
+		memmove(b->value + length, b->value, b->length);
```

Comment:
Maybe use explicit ``{}`` for the if block?

Reply:
AFAIK we don't have any guidelines on this. I'm fine with making braces mandatory for new code but I think the risks of not adding braces are largely overblown.

Reply:
I don't think it's a risk, just confusing IMHO (at least I get confused)

Reply:
The argument is usually somebody might extend the block in the `if` and forget that braces are missing.

```diff
 if (foo())
     bar();
+    baz();
```

Or `bar()` might be a macro containing multiple statements but we use `do {} while (0)` in our macros to prevent that.

Reply:
@iluuu1994 just FYI we actually have guidelines on this: https://github.com/php/php-src/blob/31692a16af0685979286b064b87e41dba987e7b1/CODING_STANDARDS.md?plain=1#L243-L258

Reply:
@bukka Ah, good to know.